### PR TITLE
refresh study Update date on change in analysis and analysis code

### PR DESCRIPTION
**Pull request by acumarav**
refresh study Update date on change in analysis and analysis code

### CODE REVIEW

**Comment by wivern** [comment]
@acumarav Could you move logic in a single place? I mean there are several event signals in different places. And when someone add a new method that modifies study then event could be missing.
I reckon hivernate event listener might be useful.

File: src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @PreUpdate
+    @PrePersist
+    @PreRemove
+    public void update(Object entity) {
+
+        if (entity instanceof Analysis) {
+            ((Analysis) entity).getStudy().setUpdated(new Date());
+        } else if (entity instanceof AnalysisFile) {
+            ((AnalysisFile) entity).getAnalysis().getStudy().setUpdated(new Date());
+        }
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java:22]: It's not obvious to me, how modified study is persisted into the database?

Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java:22]: Not sure what exactly the question
Hibernate does it, see 1.2:
https://howtodoinjava.com/hibernate/hibernate-entity-persistence-lifecycle-states/

A persistent entity has been associated with a database table row and it’s being managed by the current running Persistence Context. Any change made to such entity is going to be detected and propagated to the database during the Session flush-time

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java:22]: I meant there's no explicit call of save method for study, how to ensure it persist changes?

Comment (acumarav) [reply src/main/java/com/odysseusinc/arachne/portal/model/listeners/AnalysisChangesListener.java:22]: Study record already exists when we modify analysis, any change to study will be propagated on transaction commit
Also, it is @PrePersist event handler so we know that saving already started for the context

**Review by wivern — APPROVED** [review]

**Pull request merged**